### PR TITLE
[GECO-23] configure OCR format through UI

### PR DESCRIPTION
**Pull request by snilapwa**

### CODE REVIEW

File: cassiopeia/src/main/webapp/WEB-INF/views/admin/system_config.jsp [hunk 1]
```diff
@@ -4,30 +4,39 @@
...
+            value="${gilesAccessToken}"></form:input>
+        <small><form:errors class="error" path="gilesAccessToken"></form:errors></small>
+    </div>
+
+    <div class="page-header">
+        <h3>Type of OCR Format</h3>
+    </div>
+
+    <div class="form-group">
+        <label> <form:checkbox path="createhOCR" /> Create hOCR
```
Comment (jdamerow) [inline cassiopeia/src/main/webapp/WEB-INF/views/admin/system_config.jsp]: let's make that a drop down menu with options HOCR and Plain text

File: cassiopeia/src/main/webapp/WEB-INF/views/admin/system_config.jsp [hunk 2]
```diff
@@ -4,30 +4,39 @@
...
+    <div class="form-group">
+        <label for="gilesAccessToken">Giles Access Token</label>
+        <form:input type="text" class="form-control" id="gilesAccessToken"
+            placeholder="Token for accessing Giles" path="gilesAccessToken"
+            value="${gilesAccessToken}"></form:input>
+        <small><form:errors class="error" path="gilesAccessToken"></form:errors></small>
+    </div>
+
+    <div class="page-header">
+        <h3>Type of OCR Format</h3>
```
Comment (jdamerow) [inline cassiopeia/src/main/webapp/WEB-INF/views/admin/system_config.jsp]: Header should be "OCR Settings"

**Review by jdamerow — CHANGES_REQUESTED** [review]

File: cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java [hunk 3]
```diff
@@ -30,9 +30,15 @@
...
     
+    Map<String, String> ocrTypeMap = new HashMap<>();
+    private String plainText = "plainText";
+    private String hOCR = "hOCR";
+
     @InitBinder
     protected void initBinder(HttpServletRequest request, ServletRequestDataBinder binder, WebDataBinder validateBinder) {
         validateBinder.addValidators(new SystemConfigValidator());
+        ocrTypeMap.put(plainText, "Plain Text");
+        ocrTypeMap.put(hOCR, "HOCR");
```
Comment (jdamerow) [inline cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java]: no hardcoding of string!

File: cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java [hunk 4]
```diff
@@ -41,8 +47,16 @@ public String getConfigPage(Model model) {
...
+
+        if(propertyManager.getProperty(Properties.TESSERACT_CREATE_HOCR).equalsIgnoreCase("true")) {
+            page.setOCRType(hOCR);
+        } else {
+            page.setOCRType(plainText);
+        }
         
+        model.addAttribute("ocrTypes", ocrTypeMap);
         model.addAttribute("systemConfigPage", page);
+        model.addAttribute("selectedOCRType", page.getOCRType());
```
Comment (jdamerow) [inline cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java]: do you use that anywhere?

**Review by jdamerow — CHANGES_REQUESTED** [review]

File: cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java (hunk 4 above)
Comment (snilapwa) [reply cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java]: yes, to set this value as selected value in dropdown when page loads

Comment (snilapwa) [reply cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java]: I can use systemConfigPage for the same! removed extra parameter

**Review by jdamerow — APPROVED** [review]

**Pull request merged**